### PR TITLE
Replace `os.path` with `pathlib.Path`

### LINKED ISSUE #190: [internal] Replace`os.path` with `Pathlib`
Remove all references to `os.path` in the code and replace with the equivalents using `Pathlib` for consistency and robustness.

- [Source 1.1](https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/)
- [Source 1.2](https://treyhunner.com/2019/01/no-really-pathlib-is-great/)

### PR DESCRIPTION
**Pull request by formatc1702**
Closes #190.

### CODE REVIEW

File: src/wireviz/wireviz.py [hunk 1]
```diff
@@ -311,28 +312,28 @@ def main():
...
             sys.exit(1)
         with open_file_read(args.prepend_file) as fh:
             prepend = fh.read()
             yaml_input = prepend + yaml_input
 
     if not args.output_file:
-        file_out = args.input_file
-        pre, _ = os.path.splitext(file_out)
-        file_out = pre  # extension will be added by graphviz output function
+        file_out = file_in.parent / file_in.stem  # extension will be added by graphviz output function
```
Comment (kvid) [inline src/wireviz/wireviz.py:333]: This is a simpler and probably more readable equivalent:
```suggestion
        file_out = file_in.with_suffix('')  # Suffix will be added by graphviz output function
```

File: src/wireviz/wireviz.py [hunk 2]
```diff
@@ -311,28 +312,28 @@ def main():
...
     if args.prepend_file:
-        if not os.path.exists(args.prepend_file):
+        if not Path(args.prepend_file).exists():
             print(f'Error: prepend input file {args.prepend_file} inaccessible or does not exist, check path')
             sys.exit(1)
         with open_file_read(args.prepend_file) as fh:
             prepend = fh.read()
             yaml_input = prepend + yaml_input
 
     if not args.output_file:
```
Comment (kvid) [inline src/wireviz/wireviz.py:332]: Consider swapping the `if` and the `else` statements and remove the `not`:
```suggestion
    if args.output_file:
```

**Review by kvid — CHANGES_REQUESTED** [review]
I also like the pathlib interface. Consider the change suggested by @laurierloi and a few other code alternatives.

File: src/wireviz/wireviz.py [hunk 3]
```diff
@@ -311,28 +312,28 @@ def main():
...
+        print(f'Error: input file {file_in} inaccessible or does not exist, check path')
         sys.exit(1)
 
-    with open_file_read(args.input_file) as fh:
+    with open_file_read(file_in) as fh:
         yaml_input = fh.read()
 
     if args.prepend_file:
-        if not os.path.exists(args.prepend_file):
+        if not Path(args.prepend_file).exists():
```
Comment (kvid) [inline src/wireviz/wireviz.py:325]: For some strange reason, this suggestion was not included in my review just now:
(The same argument as @laurierloi describes above also applies here.)
```suggestion
        if not Path(args.prepend_file).is_file():
```

**Comment by formatc1702** [comment]
All comments have been addressed inside #251 or are no longer relevant after 251 is applied

**Pull request merged**